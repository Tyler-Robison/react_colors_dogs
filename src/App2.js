import React, { useState } from 'react'
import './App.css';
import RouteList from './RouteList';
import Nav from './Nav';
import colorsArr from './colors';
import useLocalStorage from './hooks/useLocalStorage';

function App2() {

  const [colors, setColors, resetColors] = useLocalStorage('colors', colorsArr)

  return (
    <div className="App">
      <h1>app 2</h1>
      <Nav colors={colors} />
      <RouteList setColors={setColors} colors={colors} />
      <button onClick={resetColors}>Reset Colors</button>
    </div>
  );
}

export default App2;