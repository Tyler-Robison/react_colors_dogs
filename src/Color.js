import { useParams, useNavigate } from 'react-router-dom';
import { useEffect } from 'react';

const Color = ({ colors }) => {

    const navigate = useNavigate();
    const { color } = useParams()

    // re-direct to color select if color isn't valid
    useEffect(() => {
        if (!colors.includes(color)) navigate('/')
    }, [])

    // console.log('color', color)
    // console.log('colors', colors)

    return (
        <div data-testid={`${color}-div`}  style={{ backgroundColor: `${color}`, height: '80vh', width: '100vh' }}>

        </div>
    )
}

export default Color;