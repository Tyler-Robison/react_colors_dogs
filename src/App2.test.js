import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Color from "./Color";
import { MemoryRouter } from "react-router-dom";
import App2 from "./App2";

// const colors = ['red', 'blue', 'green'];
// global.CSS = { supports: jest.fn() };
// clears error but test still fails

it("renders without crashing", function () {
    <MemoryRouter initialEntries={['/']}>
        <App2 />
    </MemoryRouter>
});

it("matches snapshot", function () {
    const container = render(
        <MemoryRouter initialEntries={['/']}>
            <App2 />
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

test('renders App homepage', function () {
    const { getByText } = render((
        <MemoryRouter initialEntries={['/']}>
            <App2 />
        </MemoryRouter>
    ))

    const text = getByText('Please Select A color');
    expect(text).toBeInTheDocument();
});

test('Allows us to create colors', function () {
    const app2 = render((
        <MemoryRouter initialEntries={['/']}>
            <App2 />
        </MemoryRouter>
    ))

    const text = app2.getByText('Please Select A color');
    expect(text).toBeInTheDocument();

    const link = app2.getByText('Create a new Color')
    fireEvent.click(link)

    const colorInput = app2.getByLabelText('New Color')
    const btn = app2.queryByText('Add color')
    // console.log('btn', btn)

    fireEvent.change(colorInput, { target: { value: 'pink' } })
    fireEvent.click(btn)

    // go back to homepage
    const homepageText = app2.getByText('Create a new Color');
    expect(homepageText).toBeInTheDocument();

    // color was added
    const colorText = app2.getByText('pink');
    expect(colorText).toBeInTheDocument();
});

test('Clicking on color makes correct colored div appear', function () {
    const app2 = render((
        <MemoryRouter initialEntries={['/']}>
            <App2 />
        </MemoryRouter>
    ))

    const text = app2.getByText('Please Select A color');
    expect(text).toBeInTheDocument();

    const colorText = app2.getByText('blue')
    fireEvent.click(colorText);

    const coloredDiv = app2.queryByTestId("blue-div");
    expect(coloredDiv).toBeInTheDocument();
    
});