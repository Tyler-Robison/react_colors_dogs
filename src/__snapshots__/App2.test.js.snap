// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      app 2
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/colors"
        >
          All colors
        </a>
      </li>
      <li>
        <a
          class=""
          href="/colors/red"
        >
          red
        </a>
      </li>
      <li>
        <a
          class=""
          href="/colors/blue"
        >
          blue
        </a>
      </li>
      <li>
        <a
          class=""
          href="/colors/green"
        >
          green
        </a>
      </li>
    </ul>
    <div>
      <p>
        Please Select A color
      </p>
      <a
        href="/colors/new"
      >
        Create a new Color
      </a>
    </div>
    <button>
      Reset Colors
    </button>
  </div>
</DocumentFragment>
`;
